internal/impl/postgresql/pglogicalstream/logical_stream.go

```
@@ -0,0 +1,802 @@
+	messages                   chan StreamMessage
+	snapshotName               string
+	slotName                   string
+	schema                     string
+	tableNames                 []string
+	snapshotBatchSize          int
+	decodingPlugin             DecodingPlugin
+	decodingPluginArguments    []string
+	snapshotMemorySafetyFactor float64
+	logger                     *service.Logger
+	monitor                    *Monitor
+	streamUncomited            bool
+	snapshotter                *Snapshotter
+	transactionAckChan         chan string
+	transactionBeginChan       chan bool
+
+	lsnAckBuffer []string
+
+	m       sync.Mutex
+	stopped bool
+
+	consumedCallback chan bool
+}
+
+// NewPgStream creates a new instance of the Stream struct
+func NewPgStream(ctx context.Context, config *Config) (*Stream, error) {
+	var (
+		cfg *pgconn.Config
+		err error
+	)
+
+	connectionParams := ""
+	if config.PgConnRuntimeParam != "" {
+		connectionParams = "&" + config.PgConnRuntimeParam
+	}
+
+	// intentiolly omit password to support password with special characters
+	// a new lines below cfg.Password = config.DBPassword
+	// we also need to use pgconn.ParseConfig since we are going to use this pased config to connect to the database for snapshot
+	// pgx panics when connection is not created via PaseConfig method
+	q := fmt.Sprintf("postgres://%s@%s:%d/%s?replication=database%s",
+		config.DBUser,
+		config.DBHost,
+		config.DBPort,
+		config.DBName,
+		connectionParams,
+	)
+
+	if cfg, err = pgconn.ParseConfig(q); err != nil {
+		return nil, err
+	}
+
+	cfg.Password = config.DBPassword
+	cfg.TLSConfig = config.TLSConfig
+
+	dbConn, err := pgconn.ConnectConfig(context.Background(), cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = dbConn.Ping(context.Background()); err != nil {
+		return nil, err
+	}
+
+	var tableNames []string
+	tableNames = append(tableNames, config.DBTables...)
+
+	if config.ReplicationSlotName == "" {
+		rng, _ := codename.DefaultRNG()
```

Why are we ignoring the error?